src/differentials/thunks.jl

```diff
@@ -19,6 +19,74 @@ Base.:(==)(a::AbstractThunk, b::AbstractThunk) = unthunk(a) == unthunk(b)
 Base.:(==)(a::AbstractThunk, b) = unthunk(a) == b
 Base.:(==)(a, b::AbstractThunk) = a == unthunk(b)
 
+Base.:(-)(a::AbstractThunk) = - unthunk(a)
+Base.:(-)(a::AbstractThunk, b) = unthunk(a) - b
+Base.:(-)(a, b::AbstractThunk) = a - unthunk(b)
+Base.:(/)(a::AbstractThunk, b) = unthunk(a) / b
+Base.:(/)(a, b::AbstractThunk) = a / unthunk(b)
+
+Base.real(a::AbstractThunk) = real(unthunk(a))
+Base.imag(a::AbstractThunk) = imag(unthunk(a))
+Base.Complex(a::AbstractThunk) = Complex(unthunk(a))
+Base.Complex(a::AbstractThunk, b::AbstractThunk) = Complex(unthunk(a), unthunk(b))
+
+Base.eltype(a::AbstractThunk) = eltype(unthunk(a))
+Base.convert(T, a::AbstractThunk) = convert(T, unthunk(a))
+Base.convert(::Type{T}, t::T) where T<:AbstractThunk = t
+
+Base.mapreduce(f, op, a::AbstractThunk; kws...) = mapreduce(f, op, unthunk(a); kws...)
+Base.mapreduce(f, op, itr, a::AbstractThunk; kws...) = mapreduce(f, op, itr, unthunk(a); kws...)
+Base.sum!(r, A::AbstractThunk; kws...) = sum!(r, unthunk(A); kws...)
+
+Base.size(a::AbstractThunk) = size(unthunk(a))
+Base.size(a::AbstractThunk, dim) = size(unthunk(a), dim)
+Base.vec(a::AbstractThunk) = vec(unthunk(a))
+Base.axes(a::AbstractThunk, k) = axes(unthunk(a), k)
+Base.reshape(a::AbstractThunk, args...) = reshape(unthunk(a), args...)
+Base.getindex(a::AbstractThunk, args...) = getindex(unthunk(a), args...)
+Base.setindex!(a::AbstractThunk, value, key...) = setindex!(unthunk(a), value, key...)
```

Comment:
This does not act the way people will expect it to, does it?
Maybe we should make mutating operations like this error and say that you need to unthunk first.

```
julia> k = @thunk(2*[1,2,3])
Thunk(var"#5#6"())

julia> k[1] = 100
100

julia> k[1]
2
```

Reply:
Yeah, i have changed these to error